dvc/tree/gdrive.py

```diff
@@ -510,20 +532,41 @@ def _path_to_item_ids(self, path, create, use_cache):
         )
 
     def _get_item_id(self, path_info, create=False, use_cache=True, hint=None):
-        assert path_info.bucket == self._bucket
+        # cache is not used when Tree is associated with a GDriveDependency
+        if not self.path_info:
+            use_cache = False
 
-        item_ids = self._path_to_item_ids(path_info.path, create, use_cache)
+        item_ids = self._path_to_item_ids(
+            path_info.path, create, use_cache, path_info.bucket
+        )
         if item_ids:
             return min(item_ids)
 
         assert not create
         raise FileMissingError(path_info, hint)
 
     def exists(self, path_info, use_dvcignore=True):
+        from pydrive2.files import ApiRequestError
+
+        assert isinstance(path_info, GDriveURLInfo)
         try:
-            self._get_item_id(path_info)
+            item_id = self._get_item_id(path_info)
+            file_repr = self._drive.CreateFile({"id": item_id})
+            file_repr.FetchMetadata()
+
         except FileMissingError:
             return False
+        except ApiRequestError as e:
+            error_code = e.error.get("code", 0)
+            if error_code == 404:
+                return False
+            else:
+                raise DvcException(
```

Comment:
I think we don't do this in other places right now. DVC automatically converts it upstream (probably with less details). It feels that it should be something more generic, wrapping ApiRequestError in every gdrive function probably is not very scalable. Can it be a decorator? Do we need one at all? How bad is it w/o this wrapper?

Reply:
Encapsulating it to raise a `FileMissingError` might solve everything.

Reply:
I decided to not use a decorator because it wouldn't allow me to include the file name in the `FileMissingError` message.

Reply:
I didn't use an decorator to allow adding the file name in the `FileMissingError` message.